File: media/js/mozorg/manifesto.js
/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

(function () {
    'use strict';

    // Open Twitter in a sub window
    function openTwitterSubwin(url) {
        var width = 550;
        var height = 420;
        var options = {
            scrollbars: 'yes',
            resizable: 'yes',
            toolbar: 'no',
            location: 'yes',
            width: width,
            height: height,
            top:
                screen.height > height
                    ? Math.round(screen.height / 2 - height / 2)
                    : 0,
            left: Math.round(screen.width / 2 - width / 2)
        };

        window
            .open(
                url,
                'twitter_share',
                window._SearchParams
                    .objectToQueryString(options)
                    .replace(/&/g, ',')
            )
            .focus();
    }

    function handleShareLinkClick(e) {
        var href = e.target.href;

        // Open Twitter in a sub window
        openTwitterSubwin(href);
        e.preventDefault();
    }

    function onLoad() {
        // Set up twitter link handler
        var shareLinks = document.querySelectorAll('.js-manifesto-share');

        for (var i = 0; i < shareLinks.length; i++) {
            shareLinks[i].addEventListener(
                'click',
                handleShareLinkClick,
                false
            );
        }
    }

    window.Mozilla.run(onLoad);
})();

File: tests/pages/manifesto.py
# This Source Code Form is subject to the terms of the Mozilla Public
# License, v. 2.0. If a copy of the MPL was not distributed with this
# file, You can obtain one at https://mozilla.org/MPL/2.0/.

from selenium.webdriver.common.by import By

from pages.base import BasePage


class ManifestoPage(BasePage):
    _URL_TEMPLATE = "/{locale}/about/manifesto/"

    _primary_share_button_locator = (By.CSS_SELECTOR, ".share-addendum .js-manifesto-share")
    _secondary_share_button_locator = (By.CSS_SELECTOR, ".principles-foot .js-manifesto-share")

    @property
    def is_primary_share_button_displayed(self):
        return self.is_element_displayed(*self._primary_share_button_locator)

    @property
    def is_secondary_share_button_displayed(self):
        return self.is_element_displayed(*self._secondary_share_button_locator)
